data.py

```diff
@@ -137,33 +132,133 @@ def parse_food_items(item_list, eatery_id):
+      )
+      new_dining_items.append(new_dining_item)
+    dining_items[eatery_id] = new_dining_items
+    return new_dining_items
+  except Exception:
+    return []
+
+def parse_coordinates(eatery):
+  try:
+    new_coordinates = schema.CoordinatesType(
+        latitude=eatery['coordinates']['latitude'],
+        longitude=eatery['coordinates']['longitude']
+    )
+    return new_coordinates
+  except Exception:
+    new_coordinates = schema.CoordinatesType(
+        latitude=0.0,
+        longitude=0.0
+    )
+    return new_coordinates
+
+def parse_campus_area(eatery):
+  if 'campusArea' in eatery.keys():
+    new_campus_area = schema.CampusAreaType(
+        description=eatery['campusArea']['descr'],
+        description_short=eatery['campusArea']['descrshort']
     )
-    new_dining_items.append(new_dining_item)
-  dining_items[eatery_id] = new_dining_items
-  return new_dining_items
-
-def parse_coordinates(dict):
-  new_coordinates = schema.CoordinatesType(
-      latitude=dict['latitude'],
-      longitude=dict['longitude']
-  )
-  return new_coordinates
-
-def parse_campus_area(dict):
-  new_campus_area = schema.CampusAreaType(
-      description=dict['descr'],
-      description_short=dict['descrshort']
-  )
-  return new_campus_area
+    return new_campus_area
+  else:
+    new_campus_area = schema.CampusAreaType(
+        description='',
+        description_short=''
+    )
+    return new_campus_area
+
+def parse_static_eateries(statics_json):
+  global eateries
+  for eatery in statics_json['eateries']:
+    new_id = resolve_id(eatery)
+    new_eatery = schema.EateryType(
+        about=safe_parse(eatery, 'about'),
+        about_short=safe_parse(eatery, 'aboutshort'),
+        campus_area=parse_campus_area(eatery),
+        coordinates=parse_coordinates(eatery),
+        dining_items=parse_dining_items(eatery, new_id),
+        eatery_type=parse_eatery_type(eatery),
+        id=new_id,
+        image_url=get_image_url(eatery['slug']),
+        location=safe_parse(eatery, 'location'),
+        name=safe_parse(eatery, 'name'),
+        name_short=safe_parse(eatery, 'nameshort'),
+        operating_hours=parse_static_op_hours(eatery['operatingHours'], new_id),
+        payment_methods=parse_payment_methods(eatery['payMethods']),
+        phone=parse_phone(eatery['contactPhone'], eatery['name']),
+        slug=safe_parse(eatery, 'slug')
+    )
+    eateries[new_eatery.id] = new_eatery
+
+def safe_parse(obj, key, default=''):
+  try:
+    value = obj[key]
+    if value is None:
+      return default
+    return value
+  except Exception:
+    return default
+
+def resolve_id(eatery):
+  """Returns a new id (int) for an external eatery
+  If the eatery does not have a provided id, we need to make one for it.
+  Works by going backwards from 99 and making sure the id is not already takenself.
+  Should feasibly never conflict with cornell's eateries (numbered ascending, current max is < 30).
+  """
```

Comment:
prefer triple single quotes `'''` over doubles `"""`

Reply:
@kq-li this isn't correct, docstrings are in double quotes

Reply:
@cswenberg can you redo this change?

Reply:
whoops, you're right, my bad